internal/parser/translate_httproute_test.go

```diff
@@ -0,0 +1,445 @@
+package parser
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/kong/go-kong/kong"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/kongstate"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+)
+
+func Test_fromHTTPRoutes(t *testing.T) {
+	httpPort := gatewayv1alpha2.PortNumber(80)
+	pathMatchPrefix := gatewayv1alpha2.PathMatchPathPrefix
+	pathMatchRegex := gatewayv1alpha2.PathMatchRegularExpression
+	queryMatchExact := gatewayv1alpha2.QueryParamMatchExact
+
+	for _, tt := range []struct {
```

Comment:
no testcase for an HTTPRoute with an empty `.spec.rules`

Reply:
There is already one it's on line 228:

```
    --- PASS: Test_ingressRulesFromHTTPRoutes/an_HTTPRoute_with_no_rules_can't_be_routed (0.00s)
```